From cb562ce3248357a04d67b02358b9ed73783baa6e Mon Sep 17 00:00:00 2001
From: Peng Wu <wupeng58@huawei.com>
Date: Fri, 11 Nov 2022 09:32:41 +0800
Subject: [PATCH] mm: Add reliable_nr_page for accounting reliable memory
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: cb562ce3248357a04d67b02358b9ed73783baa6e
Modified-by-SEL: No


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4SK3S
CVE: NA

--------------------------------

Adding an variable in mm_struct for accouting the amount of reliable memory
allocated by the reliable user tasks.

Signed-off-by: Peng Wu <wupeng58@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/mm_types.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/mm_types.h b/include/linux/mm_types.h
index 1c22e294f083..15ff1e20f5ca 100644
--- a/include/linux/mm_types.h
+++ b/include/linux/mm_types.h
@@ -587,7 +587,8 @@ struct mm_struct {
 #endif
 
 #ifdef CONFIG_MEMORY_RELIABLE
-		atomic_long_t reserve_0;
+		/* total used reliable pages */
+		KABI_RENAME(atomic_long_t reserve_0, atomic_long_t reliable_nr_page);
 #endif
 	} __randomize_layout;
 
-- 
2.33.0

